### PR TITLE
block (UI and API): Add mark deletion and no compaction UI

### PR DESCRIPTION
**Pull request by adzshaf**
This is part of LFX mentorship project (issue: #3112). We want to add mark deletion and no compaction button in block details UI.

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes
- Add `/block/mark` endpoint for trigger marking block API
- Add button for mark deletion and no compaction in UI, with modal for asking the detail/reason

## Verification

Tested in local using `make quickstart`

Example of mark deletion:
Example Deletion

Result of mark deletion:
image

Example of mark no compaction:
Example No Compaction

Result of mark no compaction:
image

cc: @squat @onprem

### CODE REVIEW

File: pkg/api/blocks/v1.go [hunk 1]
```diff
@@ -55,6 +63,42 @@ func (bapi *BlocksAPI) Register(r *route.Router, tracer opentracing.Tracer, logg
...
+		return nil, nil, &api.ApiError{Typ: api.ErrorBadData, Err: errors.New("Action cannot be empty")}
+	}
+
+	id, err := ulid.Parse(idParam)
+	if err != nil {
+		return nil, nil, &api.ApiError{Typ: api.ErrorBadData, Err: errors.Errorf("ULID %q is not valid: %v", idParam, err)}
+	}
+
+	switch actionParam {
+	case "DELETION":
```
Comment (bwplotka) [inline pkg/api/blocks/v1.go:88]: Can move this to Go content so we don't make typo/mistake here?

**Review by bwplotka — APPROVED** [review]
It looks pretty sweet, thank your!

I would be happy to merge it TBH, LGTM!

**Comment by bwplotka** [comment]
Do you mind rebasing to fix bindata?

File: pkg/api/blocks/v1.go [hunk 2]
```diff
@@ -55,6 +63,42 @@ func (bapi *BlocksAPI) Register(r *route.Router, tracer opentracing.Tracer, logg
...
+		err := block.MarkForDeletion(r.Context(), bapi.logger, bapi.bkt, id, detailParam, promauto.With(nil).NewCounter(prometheus.CounterOpts{}))
```
Comment (yeya24) [inline pkg/api/blocks/v1.go:89]: We don't need to use a stub counter here. I think it makes sense to use a separate counter to track how many blocks are deleted or marked manually.

**Comment by yeya24** [comment]
What about having adding this to the latest release? @bwplotka

File: pkg/api/blocks/v1.go (hunk 2 above)
Comment (yeya24) [reply pkg/api/blocks/v1.go:109]: cc @adzshaf

**Review by yeya24 — COMMENTED** [review]
Let's fix the comment, resolve the conflicts and add a changelog entry? Then we are ready to merge it I think.

**Comment by yeya24** [comment]
Just added a commit to resolve conflict. I feel this is ready to go! cc @onprem @adzshaf

File: pkg/ui/react-app/src/thanos/pages/blocks/BlockDetails.tsx [hunk 3]
```diff
@@ -11,6 +11,36 @@ export interface BlockDetailsProps {
...
+        body,
+      });
+
+      if (!response.ok) {
+        throw new Error(response.statusText);
+      }
+      setModalAction('');
+    } catch (error) {
+      setModalAction('');
+    }
```
Comment (onprem) [inline pkg/ui/react-app/src/thanos/pages/blocks/BlockDetails.tsx:41]:
```suggestion
    } finally {
      setModalAction('');
    }
```

File: pkg/api/blocks/v1.go [hunk 4]
```diff
@@ -32,8 +39,27 @@ type BlocksInfo struct {
...
+	Deletion ActionType = iota
+	NoCompaction
+	Unknown
+)
+
+func parse(s string) ActionType {
+	switch s {
+	case "DELETION":
+		return Deletion
+	case "NO COMPACTION":
```
Comment (onprem) [inline pkg/api/blocks/v1.go:54]:
```suggestion
	case "NO_COMPACTION":
```

Just a suggestion. Whitespace doesn't always play nice in bash scripts etc. so maybe it's better to keep these constants a single word.

**Review by onprem — APPROVED** [review]

**Comment by yeya24** [comment]
Sorry for my poor git skill. I rebased the code and all previous commits are also signed off by me, too. Hope this is okay @adzshaf.

I fixed the bindata and tested it working locally. Overall this looks good, but there is some work to fix:

In the dark mode, I expect the button to have a lighter color.
image

Also after I click `Mark Deletion` or `Mark No Compaction`, it works and I can see that the marker created. But in the UI I didn't get any status response to tell me whether the work succeeds or not. This is something important to have.

**Review by yeya24 — APPROVED** [review]
LGTM. Let's fix nits in next pr.

**Pull request merged**